src/pm-conversations/PmConversationList.js

```diff
@@ -24,17 +31,33 @@ type Props = $ReadOnly<{|
 /**
  * A list describing all PM conversations.
  * */
-export default class PmConversationList extends PureComponent<Props> {
+class PmConversationList extends PureComponent<Props> {
   handleUserNarrow = (email: string) => {
     this.props.dispatch(doNarrow(privateNarrow(email)));
   };
 
   handleGroupNarrow = (users: UserOrBot[]) => {
-    this.props.dispatch(doNarrow(groupNarrow(users.map(s => s.email))));
+    const { dispatch, ownUser } = this.props;
+    const emails = users.filter(u => u.user_id !== ownUser.user_id).map(u => u.email);
+    dispatch(doNarrow(groupNarrow(emails)));
+  };
+
+  /**
+   * If this is a 1:1 conversation, the other user involved. (For the
+   * self-conversation, this is the self-user.)
+   *
+   * If this is a group PM, null.
+   */
+  getOtherUser = (users: UserOrBot[]): UserOrBot | null => {
+    const ownUserId = this.props.ownUser.user_id;
+    if (users.length <= 2) {
+      return users.filter(u => u.user_id !== ownUserId)[0];
+    }
+    return null;
```

Comment:
Hmm, does this work in the self-1:1 case? It looks like it'd end up returning `[][0]`, which is `undefined`.